common/djangoapps/student/reports.py

```diff
@@ -0,0 +1,69 @@
+import unicodecsv as csv
+
+from django.utils import timezone
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from lms.djangoapps.grades.course_grade_factory import CourseGradeFactory
+
+from lms.djangoapps.grades.models import PersistentCourseGrade
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+def csv_course_report(f):
+    field_names = [
+        'Candidate RSA ID number',
+        'Candidate First Name',
+        'Candidate Last Name',
+        'Enrollment Date',
+        '% Complete to date',
+        'Completion Date',
+        'Duration between enrollment and completion',
+    ]
+    writer = csv.DictWriter(f, fieldnames=field_names)
+    for course_overview in CourseOverview.objects.all():
+        course_title = [course_overview.id, course_overview.display_name]
+        course_title.extend([''] * (len(field_names) - 2))
+        writer.writerow(dict(zip(field_names, course_title)))
+        writer.writeheader()
+
+        for enrollment in course_overview.courseenrollment_set.filter(is_active=True):
+            row_values = []
+
+            user = enrollment.user
+            row_values.append(user.username)
```

Comment:
This username is the candidate GUID  trimmed to remove the last 6 chars. The full GUID is available in the email field by removing the "@harambeecloud.com"

Reply:
@nerfani probably we have a misunderstanding - how a GUID relates to the current report specification?